corehq/apps/dropbox/tasks.py

```diff
@@ -0,0 +1,62 @@
+from celery.task import task
+from dropbox.client import DropboxClient
+from dropbox.rest import ErrorResponse
+
+from django.conf import settings
+from django.template.loader import render_to_string
+
+from dimagi.utils.django.email import send_HTML_email
+
+
+@task
+def upload(dropbox_helper_id, access_token, size, max_retries):
+    from .models import DropboxUploadHelper
+    helper = DropboxUploadHelper.objects.get(id=dropbox_helper_id)
+    client = DropboxClient(access_token)
+    retries = 0
+
+    try:
+        with open(helper.src, 'rb') as f:
+            uploader = client.get_chunked_uploader(f, size)
+            while uploader.offset < size:
+                helper.progress = uploader.offset / size
+                helper.save()
+                try:
+                    uploader.upload_chunked()
+                except ErrorResponse, e:
+                    if retries < helper.max_retries:
+                        retries += 1
+                    else:
+                        helper.failure_reason = str(e)
+                        helper.save()
+                        raise e
+
+            upload = uploader.finish(helper.dest)
+    except Exception, e:
+        helper.failure_reason = str(e)
+        helper.save()
+
+    if helper.failure_reason is None:
```

Comment:
we might also eventually want a job that goes through and purges stale files - not sure if the shared drive already has that.